File: BungiiAutomation/Bungii.Test.Regression.Android.Integration/Pages/Bungii/SearchingPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace Bungii.Test.Regression.Android.Integration.Pages
{
    class SearchingPage
    {
        public SearchingPage(IWebDriver driver)
       {
           PageFactory.InitElements(driver, this);
       }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/toolbar_button_cancel")]
        public IWebElement Link_CancelSearch { get; set; }

        [FindsBy(How = How.Id, Using ="android:id/button1")]
        public IWebElement Button_CancelConfirm { get; set; }          //By.XPath("//android.widget.Button[@text='YES']")

        [FindsBy(How = How.Id, Using = "android:id/button2")]
        public IWebElement Button_CloseCancel { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@text='SEARCHING…']")]
        public IWebElement PageTitle_Searching { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/searching_view_progressbar")]
        public IWebElement Loader { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/searching_view_subtitle")]
        public IWebElement Text_MsgSearching { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/searching_view_horizontal_progressbar")]
        public IWebElement ProgressBar { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@text='DRIVER NOT AVAILABLE']")]
        public IWebElement PageTitle_DriverNotAvailable { get; set; }
    }
}

File: Bungii.Test.Integration/src/test/java/com/bungii/android/pages/bungii/SearchingPage.java
package com.bungii.android.pages.bungii;

import com.bungii.common.core.PageBase;
import org.openqa.selenium.WebElement;

public class SearchingPage extends PageBase {

    public WebElement Link_CancelSearch () { return findElement("com.bungii.customer:id/toolbar_button_cancel", LocatorType.Id); }

    public WebElement Button_CancelConfirm () { return findElement("android:id/button1", LocatorType.Id); }          //By.XPath("//android.widget.Button[@text='YES']")

    public WebElement Button_CloseCancel () { return findElement("android:id/button2", LocatorType.Id); }

    public WebElement PageTitle_Searching () { return findElement("//android.widget.TextView[@text='SEARCHING…']", LocatorType.XPath); }

    public WebElement Loader () { return findElement("com.bungii.customer:id/searching_view_progressbar", LocatorType.Id); }

    public WebElement Text_MsgSearching () { return findElement("com.bungii.customer:id/searching_view_subtitle", LocatorType.Id); }

    public WebElement ProgressBar () { return findElement("com.bungii.customer:id/searching_view_horizontal_progressbar", LocatorType.Id); }

    public WebElement PageTitle_DriverNotAvailable () { return findElement("//android.widget.TextView[@text='DRIVER NOT AVAILABLE']", LocatorType.XPath); }
}
